README.md

```diff
@@ -434,6 +435,16 @@ const Outer = () =>
   </ColorContext.Provider>
 ```
 
+### `addMemoBoundary()`
+
+```js
+addMemoBoundary(
+  dependencies?: Array<string> | (oldProps: Object, newProps: Object) => Object
+): Function
+```
+
+Memoizes everything below it in a `flowMax` chain using [`React.memo()`](https://reactjs.org/docs/react-api.html#reactmemo). When the `dependencies` argument is an array, it will re-render whenever one of the props named in the array changes. If `dependencies` is a function, it will re-render whenever it returns `false`, indicating that `oldProps` and `newProps` are not equal. (This is equivalent to the comparison function that `React.memo()` optionally accepts.) If the `dependencies` argument is not present, it will re-render whenever the props change, using a shallow comparison of values in the props object. (This is equivalent to using `React.memo` with no custom comparison function.)
```

Comment:
I am not good at writing this stuff.

Reply:
Ha not bad

Actually I noticed that all of the docs need to explain that `dependencies` can also be a callback of `(oldProps, newProps) => boolean` (your return type is wrong there). So I opened #52 and I'd say let's not even document the callback style yet, and I can go through separately and think about how to maybe DRY-ly explain that all dependencies args also can be callbacks? (also noticed that the boolean return value is the _opposite_ for `addMemoBoundary()` vs others (eg `addEffect()`) - this mimics how `React.memo()` expects an opposite boolean return value vs eg Recompose's [`withPropsOnChange()`](https://github.com/acdlite/recompose/blob/master/docs/API.md#withpropsonchange), so probably what we want, but kind of annoying/unintuitive)

So how about:
```
Avoids unnecessary re-rendering of everything below it in a `flowMax` chain

Wraps [`React.memo`](https://reactjs.org/docs/react-api.html#reactmemo)

The optional argument is an array of names of props that should trigger re-rendering whenever one of those props changes (based on `===` comparison). If the argument is omitted, re-rendering will occur whenever _any_ prop changes (like `React.memo()`'s default behavior)
```

And then refine the examples a little bit (omit the callback comparison for now, use comments to explain the behavior, indentation):
```
const DefaultDoubler = flow(
  addMemoBoundary(), // this component will re-render whenever any of its props changes
  addProps(({num}) => ({doubled: num * 2})),
  ({doubled}) =>
    <div>Doubled: {doubled}</div>
)

const PropsArrayDoubler = flow(
  addMemoBoundary(["num"]), // this component will re-render whenever its `num` prop changes
  addProps(({num}) => ({doubled: num * 2})),
  ({doubled}) =>
    <div>Doubled: {doubled}</div>
)
```